deployment/ccip/1_6_0/sequences/mcms.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/Masterminds/semver/v3"
+	"github.com/smartcontractkit/chainlink-ccip/deployment/deploy"
+	"github.com/smartcontractkit/chainlink-ccip/deployment/utils/sequences"
+	cldf_chain "github.com/smartcontractkit/chainlink-deployments-framework/chain"
+	"github.com/smartcontractkit/chainlink-deployments-framework/chain/ton"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/smartcontractkit/chainlink-ton/deployment/ccip/helpers"
+	mcmsConfig "github.com/smartcontractkit/chainlink-ton/deployment/mcms/config"
+	mcmsSeq "github.com/smartcontractkit/chainlink-ton/deployment/mcms/sequence"
+	"github.com/smartcontractkit/chainlink-ton/deployment/state"
+	"github.com/smartcontractkit/chainlink-ton/deployment/utils/sequence"
+	"github.com/smartcontractkit/chainlink-ton/pkg/ton/tvm"
+	"github.com/xssnick/tonutils-go/address"
+)
+
+const defaultMCMSContractCoin = "1.5"
+
+func (a *TonAdapter) DeployMCMS() *operations.Sequence[deploy.MCMSDeploymentConfigPerChainWithAddress, sequences.OnChainOutput, cldf_chain.BlockChains] {
+	return DeployMCMSContracts
+}
+
+var DeployMCMSContracts = operations.NewSequence(
+	"deploy-mcms",
+	semver.MustParse("0.0.4"), // TODO mcms and timelock has different versions, can we pick mcms version here?
+	"Deploys all MCM contracts with config",
+	func(b operations.Bundle, chains cldf_chain.BlockChains, input deploy.MCMSDeploymentConfigPerChainWithAddress) (output sequences.OnChainOutput, err error) {
+		var txs [][]byte
+
+		tonChain := chains.TonChains()[input.ChainSelector]
+		deps, err := extractTonDepsFromMCMSDeploymentInput(tonChain)
+		if err != nil {
+			return sequences.OnChainOutput{}, err
+		}
+		seqInput := intoDeployMCMSSeqInput(input, deps.TonChain.WalletAddress)
+		mcmsSeqReport, err := operations.ExecuteSequence(b, mcmsSeq.DeployMCMSSequence, deps, seqInput)
+		if err != nil {
+			return sequences.OnChainOutput{}, fmt.Errorf("failed to deploy MCMS for TON chain %d: %w", input.ChainSelector, err)
+		}
+
+		txs = append(txs, mcmsSeqReport.Output.Transactions...)
+
+		// Execute the txs || MCMS proposals
+		err = helpers.ExecuteTransactions(b.GetContext(), b.Logger, deps.TonChain.Client, deps.TonChain.Wallet, txs)
+		if err != nil {
+			return sequences.OnChainOutput{}, err
+		}
+
+		return sequences.OnChainOutput{}, nil
+	},
+)
+
+func extractTonDepsFromMCMSDeploymentInput(chain ton.Chain) (mcmsConfig.MCMSDeps, error) {
+	deps := mcmsConfig.MCMSDeps{
+		TonChain: chain,
+		MCMSChainState: map[uint64]state.MCMSChainState{
+			// initialize with zero addresses; actual addresses will be filled in after deployment
+			chain.Selector: {
+				Timelock: *tvm.ZeroAddress,
+				MCMS:     *tvm.ZeroAddress,
+			},
+		},
+	}
+	return deps, nil
+}
+
+func intoDeployMCMSSeqInput(cfg deploy.MCMSDeploymentConfigPerChainWithAddress, deployer *address.Address) mcmsSeq.DeployMCMSSeqInput {
+	// The external config uses mcmstypes.Config which has signers, but assumes all evm address for, but
```

Comment:
@krebernisak Question about signer type. Here I'm following the existing behavior of using deployer address for initialization. 

But I wonder if we want to support input config at one point, the init roles address are TON address type, but the MCMSConfig from the api input uses EVM address.